src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -336,162 +350,223 @@ public Mono<TransactionInfoDto> updateTransactionAuthorization(
+                                                                                                                )
+                                                                                                                .toList()
+                                                                                                )
+                                                                                                .status(
+                                                                                                        TransactionStatusDto
+                                                                                                                .fromValue(
+                                                                                                                        transactionDocument
+                                                                                                                                .getStatus()
+                                                                                                                                .toString()
+                                                                                                                )
+                                                                                                )
+                                                                                );
+                                                                    }
+                                                            )
+
+                                            )
+                                    );
+                        }
+                );
+
+    }
+
+    private Mono<Either<TransactionInfoDto, Mono<BaseTransaction>>> analyzeTransactionStatus(
+                                                                                             TransactionId transactionId
+    ) {
+        Mono<BaseTransaction> baseTransaction = transactionsUtils.reduceEvents(transactionId);
+        /*
+         * Searching the transaction with authorization completed event. If none is
+         * found then returning Either.right for processing continuation, otherwise
+         * return the completed OK response taking information from the current reduced
+         * transaction status. The check is performed directly on the authorization
+         * completed event and not on the fact that the reduced transaction is instance
+         * of BaseTransactionWithCompletedAuthorization because the transaction can go
+         * in REFUND or EXPIRED statuses and their aggregates does not directly extend
+         * the BaseTransactionWithCompletedAuthorization just because transaction can
+         * EXPIRE in any state and can be REFUND also if the authorization was only
+         * requested so those aggregates. So a check is performed against the
+         * TRANSACTION_AUTHORIZATION_COMPLETED_EVENT event
+         */
```

Comment:
This method does a little too many things, I would refactor this to be a `Mono<Boolean> wasTransactionAuthorizationCompleted(TransactionId transactionId)` and let whoever calls this do any further processing

Also: Eclipse formatter _hates_ block comment, use `@formatter:off` and `@formatter:on` at the start and end of comments so that formatting doesn't get murdered :)

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/168/commits/df1075f05015a02f1bc42219402ce1a7816cca9a